WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -210,6 +250,11 @@ private async Task FailTransactionSigningPhaseAsync(Round round, CancellationTok
 				Prison.Note(alice, round.Id);
 			}
 
+			foreach (var alice in AlicesById.Select(x => x.Value))
+			{
+				RemoveAlice(alice);
+			}
```

Comment:
`ConcurrentDictionary` does allow concurrent iteration and modifications, but I'm not sure if the semantics are clear (i.e. might the iteration miss some elements).

As for what this code does, uh... LOL. It's completely wrong ;-) I don't know how I confused myself so much but it should iterate through `round.Alices`.

This is something I really hate about how blame rounds are created, instead of just passing in the list of allowed `Outpoint`s, the corresponding `Alice`s are *not* removed from the Round. I don't know why I didn't change this too, it should just pass a normal parameter instead of relying on indirect passing of parameters through mutable state...